dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -819,7 +819,21 @@ ShowStatusPropertyDialog(
         else if (pProperties->Status == NCS_MEDIA_DISCONNECTED || pProperties->Status == NCS_DISCONNECTED ||
                  pProperties->Status == NCS_HARDWARE_DISABLED)
         {
-            ShowNetConnectionProperties(pContext->pNet, pContext->hwndDlg);
+            WCHAR buffer[100];
+            
+            LoadStringW(netshell_hInstance, IDS_NETWORKCONNECTION, buffer, _countof(buffer));
+            HWND prophwnd = FindWindow(NULL, buffer);
+
+            /* if the window is already open, prevent it from opening again */
+            if (prophwnd != NULL)
+            {
+                ShowWindow(prophwnd, SW_SHOWNORMAL);
+                SetForegroundWindow(prophwnd);
+            }
+            else
+            {
+                ShellExecuteW(NULL, L"open", L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}", NULL, SW_SHOWNORMAL);
```

Comment:
Can you ShellExecuteEx invokeidlist the pidl you get from CSIDL_CONNECTIONS instead?

Reply:
I tried to do it by doing this, and I have absolutely no idea what I'm getting wrong because this works with other CSIDL's, but not with CSIDL_CONNECTIONS.

```
LPITEMIDLIST pidl = NULL;
if (SUCCEEDED(SHGetSpecialFolderLocation(NULL, CSIDL_CONNECTIONS, &pidl)))
{
    SHELLEXECUTEINFOW shExInfo = { 0 };
    shExInfo.cbSize = sizeof(shExInfo);
    shExInfo.fMask = SEE_MASK_INVOKEIDLIST;
    shExInfo.hwnd = NULL;
    shExInfo.lpVerb = L"open";
    shExInfo.lpIDList = pidl;
    shExInfo.nShow = SW_SHOWNORMAL;

    ShellExecuteExW(&shExInfo);
}
```

I don't know if I'm getting Tunnel Vision and there's some other way to approach this that you guys will be able to find faster than me (I'm still only getting started out with coding for ReactOS), but I tried my best.

Reply:
Yes, it does appear that CSIDL_CONNECTIONS is broken. Your code is correct except that it leaks the PIDL. Do your original workaround and I will fix this later.

Reply:
@Connierollstheball Does your code snippet (with SHGetSpecialFolderLocation) work on actual Windows? XP/2003 ? Vista+?

Reply:
@HBelusca Yes it works on Windows. ROS does not handle CSIDL_CONNECTIONS at all for some reason.